### PR TITLE
eliminate unused configuration

### PR DESCRIPTION
**Pull request by martyall**
## Changes

This PR removes several "unused" configuration options. Unused can mean one of two things:

1. The option is literally unused, meaning it doesn't appear anywhere outside the definition of the `Node_config` module
2. All existing build profiles use the same value, so there is no point in inspecting this value at runtime.

In the event of (2) above, I also performed obvious rewriting rules to eliminate dead code.

I will list the removed configuration options by category. The only one of these seemed slightly controversial to me is `generate_genesis_proof`, not sure how often we toggle that one.

### Category (1)

- [debug_logs](https://github.com/MinaProtocol/mina/pull/15871/commits/31bcda990ade9f3ff29fa1f36f973dcbe40cbfca)
- [record_async_backtraces](https://github.com/MinaProtocol/mina/pull/15871/commits/222bf8f847426283b7e3200ee33ac64d20ce4da9)
- [time_offsets](https://github.com/MinaProtocol/mina/pull/15871/commits/41010e5697a809b66f5c1df593946447a7abea84)
- [integration_tests](https://github.com/MinaProtocol/mina/pull/15871/commits/8e86b130f22f8cb9dec1c289d6ba494dee7fc932)
- [forced_updates](https://github.com/MinaProtocol/mina/pull/15871/commits/4f6f038cb65801e60688a57fc8bcabd2edcc29aa)

### Category (2)

- [generate_geneis_proof](https://github.com/MinaProtocol/mina/pull/15871/commits/e61b7497ad243e16f6bed256b5a6cf7e230380cb)
- [call_logger](https://github.com/MinaProtocol/mina/pull/15871/commits/31bcda990ade9f3ff29fa1f36f973dcbe40cbfca)
- [download_snark_keys](https://github.com/MinaProtocol/mina/pull/15871/commits/02b4de86f86f1df2dc82cd7fdf3d74422aa5dc6d)

(related to #15538)

### CODE REVIEW

**Comment by martyall** [comment]
!ci-build-me

**Comment by martyall** [comment]
!ci-build-me

**Comment by martyall** [comment]
!ci-build-me

File: src/lib/precomputed_values/gen_values/gen_values.ml [hunk 1]
```diff
@@ -210,70 +130,11 @@ let main () =
...
-              [%e
-                match compiled_values with
-                | Some
-                    { proof_data = Some { blockchain_proof_system_id = id; _ }
-                    ; _
-                    } ->
-                    [%expr Some [%e vk_id_to_expr ~loc id]]
-                | _ ->
-                    [%expr None]]
+          ; constraint_system_digests = [%e [%expr None]]
```
Comment (mrmr1993) [inline src/lib/precomputed_values/gen_values/gen_values.ml:133]: Nit: `= None` here and below

**Review by mrmr1993 — APPROVED** [review]
Fab!

**Pull request merged**